Sources/Player/PlaybackEngine/Internal/Asset.swift

```diff
@@ -50,16 +49,16 @@ open class Asset {
 		self.playbackMetadata = playbackMetadata
 	}
 
-	/// Method to obtain the current caching status
-	/// - Returns: The caching status
-	public func isCached() -> Bool {
-		cached
+	/// Method to get the current caching state
+	/// - Returns: The asset cache state
+	public func getCacheState() -> AssetCacheState? {
```

Comment:
[Question] when could this be optional and when would we actually like for it to be optional instead of not cached?

Reply:
Optional will be when we don't want to cache it. (It is a stored file or some format/quality we don't want cached)
That logic is done at a higher level (`InternalPlayerLoader`) to avoid propagating extra logic to the player that is not needed.